### DISCUSSION TITLE
How to access index beyond [-2]?

### DISCUSSION THREAD

**Discussion by CadeHalcyon** [kernc/backtesting.py#836] [General]
Hi, how do we access info from Data more then [-2] back, please?  See below simple example for illustrative purposes:

`def next(self):

          price = self.data.Close[-1]
          Signal = self.Signal[-1]
          Pre_Signal_1 = self.Signal[-2]
          Pre_Signal_2 = self.Signal[-3]`

When I try this, I get the following error regarding Pre_Signal_2:

`IndexError:  index -3 is out of bounds for axis 0 with size 2`

**Comment by kernc** [comment]
You can simply wrap it all in:
```py
max_lookback = 3
if len(self.Signal) > max_lookback:
    ...
```

**Reply by CadeHalcyon** [reply]
Thanks!  Your answer also helped me understand _why_ I was getting the error too (I'm relatively new to coding).
